src/libfsm/walk2.c

```diff
@@ -0,0 +1,469 @@
+#include "walk2.h"
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <errno.h>
+
+#include <adt/set.h>
+
+#include <fsm/fsm.h>
+#include <fsm/out.h>
+#include <fsm/options.h>
+#include <fsm/bool.h>
+#include <fsm/pred.h>
+#include <fsm/walk.h>
+
+#include "internal.h"
```

Comment:
"walk2.h" goes here, since it's internal

Reply:
I had that originally, but it's too easy to mask include dependencies.  This guarantees a compiler error if walk2.h requires includes that it doesn't provide.

Reply:
Yes, but it goes here.